soc/arm/nxp_s32/s32ze/Kconfig.defconfig.series

```diff
@@ -0,0 +1,34 @@
+# NXP S32Z/E MCUs family default configuration
+
+# Copyright 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_SERIES_S32ZE_R52
+
+config SOC_SERIES
+	default "s32ze"
+
+config NUM_IRQS
+	# must be >= the highest interrupt number used
+	default 991
+
+config FPU
+	default y
+
+config MAX_DOMAIN_PARTITIONS
+	default 24 if USERSPACE
```

Comment:
Can I ask how many MPU regions your soc have? Do you enable `CONFIG_MPU_GAP_FILLING`? If it is enabled and suppose the regions the soc supported is 32, I guess the `MAX_DOMAIN_PARTITIONS` should be small than 24. Because when you insert a region into the middle of the existing one, you possibly get 2 more regions? That is 24 partitions can cause 40+ regions with `CONFIG_MPU_GAP_FILLING` enabled. It's just a concern, I am okay with it if it turns out to be not an issue :)

Reply:
Hmm thanks for the heads up, I needed to revisit this part. The MPU has 20 regions for EL1 and `CONFIG_MPU_GAP_FILLING` is enabled. Leaving `MAX_DOMAIN_PARTITIONS` to the default of 16 seems a better option.